pandas/tests/extension/decimal/array.py

```diff
@@ -67,8 +67,11 @@ class DecimalArray(OpsMixin, ExtensionScalarOpsMixin, ExtensionArray):
 
     def __init__(self, values, dtype=None, copy=False, context=None):
         for i, val in enumerate(values):
-            if is_float(val) and np.isnan(val):
-                values[i] = DecimalDtype.na_value
+            if is_float(val):
```

Comment:
why is this changed?

Reply:
See here: https://github.com/pandas-dev/pandas/pull/44722#issuecomment-984957259

When deserializing from JSON, `_from_sequence` is called with a numpy array of strings, which in turn was not supported by the `__init__` function.